salt/cloud/clouds/vmware.py

```diff
@@ -977,6 +977,66 @@ def _valid_ip(ip_address):
     return True
 
 
+def _valid_ip6(ip_address):
```

Comment:
This will return true for link-local and localhost addresses, which is not the behaviour we need here. (We only need routable addresses). Using ipaddress salt._compat though would be a good idea, so I'll convert this function to use this instead of manually parsing the address

Reply:
you can use the is_private function in salt.utils.network as well, and chain them together for the functionality that you want.

`if not __utils__['network.is_private'](ip_address) and __utils__['network.is_ipv6'](ip_address):`